prowler/lib/outputs/json.py

```diff
@@ -104,137 +111,184 @@ def generate_json_asff_status(status: str) -> str:
-                        finding.check_metadata.Remediation.Code.NativeIaC,
-                        finding.check_metadata.Remediation.Code.Terraform,
-                        finding.check_metadata.Remediation.Code.CLI,
-                        finding.check_metadata.Remediation.Code.Other,
-                        finding.check_metadata.Remediation.Recommendation.Url,
-                    ],
+        compliance = Compliance_OCSF(
+            status=generate_json_ocsf_status(finding.status),
+            status_detail=finding.status_extended,
+            requirements=unroll_dict_to_list(
+                get_check_compliance(
+                    finding, finding.check_metadata.Provider, output_options
                 )
             ),
-            desc=finding.check_metadata.Remediation.Recommendation.Text,
-        ),
-        types=finding.check_metadata.CheckType,
-        src_url=finding.check_metadata.RelatedUrl,
-    )
-    finding_output.resources.append(
-        Resources(
-            group=Group(name=finding.check_metadata.ServiceName),
-            region=resource_region,
-            name=resource_name,
-            labels=resource_labels,
-            uid=resource_uid,
-            type=finding.check_metadata.ResourceType,
-            details=finding.resource_details,
         )
-    )
-    finding_output.time = timestamp.isoformat()
-    finding_output.metadata = Metadata(
-        product=Product(
-            feature=Feature(
-                uid=finding.check_metadata.CheckID,
-                name=finding.check_metadata.CheckID,
-            )
-        ),
-        original_time=timestamp.isoformat(),
-        profiles=[audit_info.profile]
-        if hasattr(audit_info, "organizations_metadata")
-        else [],
-    )
-    finding_output.compliance = Compliance_OCSF(
-        status=finding_output.status,
-        status_detail=finding_output.status_detail,
-        requirements=unroll_dict_to_list(
-            get_check_compliance(
-                finding, finding.check_metadata.Provider, output_options
-            )
-        ),
-    )
+        finding_output = Check_Output_JSON_OCSF(
+            finding=finding_ocsf,
+            resources=resources,
+            status_detail=finding.status_extended,
+            message=finding.status_extended,
+            severity=finding.check_metadata.Severity.capitalize(),
+            severity_id=generate_json_ocsf_severity_id(finding.check_metadata.Severity),
+            status=generate_json_ocsf_status(finding.status),
+            status_id=generate_json_ocsf_status_id(finding.status),
+            compliance=compliance,
+            cloud=cloud,
+            time=timestamp.isoformat(),
+            metadata=metadata,
+        )
+        return finding_output
+    except Exception as error:
+        logger.error(
+            f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+        )
+
+
+def generate_json_ocsf_status(status: str):
```

Comment:
Please include tests for this new functions. Thanks!

- `generate_json_ocsf_status`
- `generate_json_ocsf_status_id`
- `generate_json_ocsf_severity_id`

Reply:
done